crates/miden-lib/asm/agglayer/account_components/agglayer_faucet.masm

```diff
@@ -0,0 +1,137 @@
+use.agglayer::bridge_in
+use.miden::active_account
+use.miden::active_note
+use.miden::contracts::faucets
+use.miden::note
+use.miden::tx
+
+# CONSTANTS
+# =================================================================================================
+
+# The slot in this component's storage layout where the bridge account ID is stored.
+const BRIDGE_ID_SLOT = word("miden::agglayer::faucet")
+
+const P2ID_NOTE_NUM_INPUTS = 2
+const OUTPUT_NOTE_TYPE_PUBLIC = 1
+const OUTPUT_PUBLIC_NOTE_INPUTS_ADDR = 8
+const OUTPUT_PUBLIC_NOTE_INPUTS_LEN_MEM_ADDR = 0
+const P2ID_SCRIPT_ROOT = [15783632360113277539, 7403765918285273520, 15691985194755641846, 10399643920503194563]
+
+# ERRORS
+# =================================================================================================
+
+const ERR_CLAIM_INVALID_GER_PROOF = "invalid ger claim proof"
+const ERR_CLAIM_WRONG_NUMBER_OF_INPUTS = "CLAIM note must have exactly 12 inputs"
+
+#! Validates a CLAIM note by verifying its inputs against a GER against the AggLayer bridge.
+#!
+#! This procedure performs a foreign procedure invocation (FPI) to the configured
+#! AggLayer bridge account to verify its claim data (asset, amount, recipient) inclusion against
+#! one of the Global Exit Roots in the bridge storage.
+#!
+#! Inputs:  []
+#! Outputs: []
+#!
+#! Panics if:
+#! - the bridge account ID is not properly configured in storage.
+#! - the foreign procedure invocation fails.
+#! - the claim proof validation fails.
+#!
+#! Invocation: exec
+proc validate_claim_via_fpi
+    # get check_claim_proof root
+    procref.bridge_in::check_claim_proof
+    # => [MAST_ROOT]
+
+    push.BRIDGE_ID_SLOT[0..2]
+    # => [bridge_id_idx, MAST_ROOT]
+
+    # get bridge account id
+    exec.active_account::get_item
+    # => [bridge_account_id_prefix, bridge_account_id_suffix, 0, 0, MAST_ROOT]
+
+    movup.2 drop movup.2 drop
+    # => [bridge_account_id_prefix, bridge_account_id_suffix, MAST_ROOT]
+
+    # call check_claim_proof proc on bridge
+    exec.tx::execute_foreign_procedure
+    # => [validation_result]
+
+    assert.err=ERR_CLAIM_INVALID_GER_PROOF drop
+    # => []
+end
+
+#! Validates a CLAIM note against the AggLayer bridge and mints the corresponding asset to the recipient.
+#!
+#! This procedure validates the rollup exit root Merkle Proof via FPI against the agglayer bridge,
+#! and if validation passes, mints the asset and creates an output note for the recipient.
+#!
+#! Inputs:  [pad(16)]
+#! Outputs: [pad(16)]
+#!
+#! Panics if:
+#! - the rollup exit root Merkle Proof validation via FPI fails.
+#! - any of the validations in faucets::distribute fail.
+#!
+#! Invocation: call
+pub proc claim
```

Comment:
As mentioned in the previous comment, I'd probably pass the inputs into this procedure explicitly. This will help us define the interface boundaries more cleanly. I believe the inputs are:

- Info about the claim: recipient account ID, amount (probably as a u256 values), probably bridge leaf indexes. The point of this info would be twofold:
  - To make sure we can compute hash of the bridge message.
  - To deterministically derive the `P2ID` note's serial number.
- Global exit tree Merkle proof (passed in as a commitment).
- Probably some info about GER - so that we know which GER to verify the proof against. This could probably be just the GER index in the L1InfoTree.